### PR TITLE
[UX] Tab Icons

### PR DESCRIPTION
**Pull request by Teranode**
The way the tabs are currently used seems to be fine with me. There is this issue here: https://github.com/octobercms/october/issues/3856

Follwoing @LukeTowers advice the way it works is:

```
tabs:
     icons:
          'tab name here': 'icon-class-here'
     fields:
         test:
             label: test
             tab: 'tab name here'
````

also added it in the docs: https://github.com/octobercms/docs/pull/330

### CODE REVIEW

File: modules/backend/widgets/form/partials/_form_tabs.htm [hunk 1]
```diff
@@ -14,7 +14,9 @@
 <div class="<?= $navCss ?>">
     <ul class="nav nav-tabs">
         <?php $index = 0; foreach ($tabs as $name => $fields): ?>
-            <li class="<?= $index++==0?'active':''?>"><a href="#<?= $type.'tab-'.$index ?>"><?= e(trans($name)) ?></a></li>
+            <li class="<?= $index++==0?'active':''?>">
+                <a href="#<?= $type.'tab-'.$index ?>"><?php if ($tabs->getIcon($name)): ?><i class="oc-<?= $tabs->getIcon($name) ?>"></i><?php endif; ?><?= e(trans($name)) ?></a>
```
[the remark this answers is not in the document]
Comment (Teranode) [reply modules/backend/widgets/form/partials/_form_tabs.htm]: removing oc- removes the margin it creates to separate the text

**Comment by Teranode** [comment]
@LukeTowers okay i made revisions and replied to your comment

File: modules/backend/widgets/form/partials/_form_tabs.htm (hunk 1 above)
[the remark this answers is not in the document]
Comment (LukeTowers) [reply modules/backend/widgets/form/partials/_form_tabs.htm]: Could you remove it anyways and re-add right margin only through CSS?

[the remark this answers is not in the document]
Comment (LukeTowers) [reply modules/backend/widgets/form/partials/_form_tabs.htm]: @Teranode remove the oc- prefix but then add a right margin if required with CSS

[the remark this answers is not in the document]
Comment (Teranode) [reply modules/backend/widgets/form/partials/_form_tabs.htm]: @LukeTowers just did, can you check to see if that what you want?

**Comment by ayumihamsaki** [comment]
@LukeTowers @Teranode Hi, I got a suggestion with regards to this (maybe it is useful or not?). Would it be a good idea to add `position` to the YAML file?

What I mean is the following example:

```
tabs:
     icons:
          'tab name here': 'icon-class-here'
          position: left
     fields:
         test:
             label: test
             tab: 'tab name here'
```

# Syntax:

`Position: left|top|right|bottom`

For example:

# Top

`<li><a href="#contentTabOne"><i class="icon-diamond"></i><br>Text</a></li>`

# Left

`<li><a href="#contentTabOne"><i class="icon-diamond"></i> Text</a></li>`

# Right

`<li><a href="#contentTabOne">Text <i class="icon-diamond"></i></a></li>`

# Bottom

`<li><a href="#contentTabOne">Text<br><i class="icon-diamond"></i></a></li>`

Hopefully makes sense what I mean in a crude way.

**Comment by LukeTowers** [comment]
@ayumihamsaki hmm, I'm going to say no, that wouldn't be that useful. Tabs with icons located above or below wouldn't look or work well, and left vs right for RTL languages could be accomplished with just CSS.

**Comment by Teranode** [comment]
@LukeTowers so which makes sense:

```
> span[class*="icon-margin"] {
    margin-right: 8px;
}
```

or 
```
> span:not([class*="oc-"]) {
    margin-right: 8px;
}    
```

**Comment by LukeTowers** [comment]
@Teranode I would go with just targeting the span itself. If the oc-icon class is used, then it will automatically override the style applied here anyways wouldn't it?

**Pull request merged**